src/modules/firebase/tariff-zones.ts

```diff
@@ -0,0 +1,27 @@
+import { getFirestore, getDoc, doc } from 'firebase/firestore';
+import app from './firebase';
+import { TariffZone, tariffZoneSchema } from './types';
+
+export async function getTariffZones() {
+  const firestore = getFirestore(app);
+  const docRef = doc(firestore, 'configuration', 'referenceData');
+
+  const snapshot = await getDoc(docRef);
+
+  if (snapshot.exists()) {
+    const data = snapshot.data();
+    const tariffZonesFromFirebase = data.tariffZones;
+
+    const potential = JSON.parse(tariffZonesFromFirebase);
```

Comment:
Can this fail?

Reply:
Hm, it could probably. The document is a JSON object, but that could be formatted wrong of course

Reply:
I've added a try catch to the parsing